Something/WebApplication/wwwroot/js/apps/common/GlobalService.ts

```diff
@@ -0,0 +1,25 @@
+import { PageTypes } from "./PageTypeEnum";
+import { IAuthenticationProps } from "../authentication/types/IAuthenticationProps";
+import { BaseParams } from "./types/BaseParams";
+import { IOnboardingProps } from "../onboarding/types/IOnboardingProps";
+
+export const getParams = (pageType: PageTypes): BaseParams => {
+    let params = null;
+
+    switch (pageType) {
+        case PageTypes.LOGIN:
+        case PageTypes.REGISTER:
+            {
+                params = (window as any).authParams as IAuthenticationProps;
+                break;
+            }
+        case PageTypes.ONBOARDING:
+            {
+                params = (window as any).onboardingParams as IOnboardingProps;
+                break;
+            }
+        default:
+            break;
+    }
```

Comment:
I would separate this into different methods.

Reply:
Like `getOnboardingPageParams()`, `getAbcPageParams()` and then they can return the specific interface?

Reply:
correct.